parser/c2_parser_stmt.c2

```diff
@@ -390,10 +391,36 @@ fn Stmt* Parser.parseAssertStmt(Parser* p) {
     SrcLoc loc = p.tok.loc;
     p.consumeToken();
     p.expectAndConsume(Kind.LParen);
+    SrcLoc eloc = p.tok.loc;
     Expr* inner = p.parseExpr();
+    u32 elen = p.prev_loc - eloc;
     p.expectAndConsume(Kind.RParen);
+    u32 loc_end = p.prev_loc;
     p.expectAndConsume(Kind.Semicolon);
-    return p.builder.actOnAssertStmt(loc, inner);
+
+    Expr* call = nil;
+    if (p.has_asserts) {
+        // add c2_assert.c2_assert_fail func designator
+        Ref[2] ref;
+        ref[0].loc = loc;
+        ref[0].name_idx = p.pool.addStr("c2_assert", true);
+        ref[1].loc = loc;
+        ref[1].name_idx = p.pool.addStr("c2_assert_fail", true);
```

Comment:
adding c2_assert / c2_assert_fail should only be added once and stored somewhere, like we do with other constant names.

Reply:
Done.